parachain/primitives/core/src/nft.rs

```diff
@@ -0,0 +1,46 @@
+use codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use serde_derive::{Serialize, Deserialize};
+use frame_support::dispatch::{DispatchError, DispatchResult};
+use sp_runtime::RuntimeDebug;
+use sp_core::H160;
+use sp_runtime::sp_std::prelude::Vec;
+
+use artemis_ethereum::U256;
+
+/// Token info
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub struct TokenInfo<AccountId, Data> {
+	/// Token owner
+	pub owner: AccountId,
+	/// Token metadata
+	pub metadata: Vec<u8>,
+	/// Token Properties
+	pub data: Data,
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub struct ERC721TokenData {
```

Comment:
This decision is something I'm not entirely happy with: the defintion of this data struct as a global one, then injected in NFT's pallet runtime instance as a parameter.
This is a consequence of parameterizing the NFT pallet over the TokenData, which I think is a cool idea (taken from ORML). Both ERC721 and the concrete NFT pallet are necessarily aware of the type in the data.

One alternative approach is to get rid of this TokenData in the nft pallet (or just use `()`) and instead use a map in the ERC721 to store the necessary data. This could be seen as cleaner.
However, I chose this implementation because it provides a more general NFT pallet that can be used differently. However, I believe you cannot instance more than once a pallet in the runtime, so this benefit would be hindered from this limitation.